.github/workflows/rust.yml

```diff
@@ -80,3 +80,47 @@ jobs:
           DO_FEATURE_MATRIX: true
         run: ./contrib/test.sh
 
+  wasm_test_macos:
+    name: WebAssembly Test
+    strategy:
+      matrix:
+        rust:
+          - stable
+        target: [ x86_64-unknown-linux-gnu, x86_64-apple-darwin ]
+        include:
+
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+            ci_cc: clang-12
+            ci_ar: ar
```

Comment:
These are the defaults yeah? (or at least the default clang can compile to WASM I presume)

I think it would be easier to understand and more maintainable if we integrate this wasm build into the `wasm` job on line 33.

You can easily install llvm and set all the required things in a single step using this: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable

That would make it more obvious what needs to be done make things work on MacOS.